### DISCUSSION TITLE
How to display data property in API response from createError?

### DISCUSSION THREAD

**Discussion by MohanDrey24** [nitrojs/nitro#2882] [Help Wanted]
I have an error handler that throws a createError:

```
  if (e) {
    throw createError({
      status: 400,
      statusMessage: "Bad Request",
      message: e.message,
      data: "hello world",
    });
```

Why does data property doesn't show in the API response? I tried hard coding it with a string but it still doesn't show up in my response.

**Comment by drewbaker** [comment]
I would like an answer to this too. Also, how to force the error to be a JSON object, not a HTML page.

**Comment by pi0** [comment]
Please check https://github.com/nitrojs/nitro/pull/3002, in next version, `data` object will be included and production errors are json by default (unless a wrapper framework like nuxt overrides to SSR/HTML responses)
